### ISSUE TITLE
Tune trial-resources is named trial_resources in code

### ISSUE THREAD

**Issue by RoeeEnbar** [ray-project/ray#2393]
### System information
- **OS Platform and Distribution**:
Linux Ubuntu 16.04
- **Ray installed from (source or binary)**:
binary
- **Ray version**:
0.5.0
- **Python version**:
2.7.14
- **Exact command to reproduce**:
Add the following to tune/examples/pbt_example.py:
```
    run_experiments(
        {
            "pbt_test": {
                "run": "my_class",
                "trial-resources": {"cpu": 4, "gpu": 1},
                "stop": {
                    "training_iteration": 2 if args.smoke_test else 99999
                },
                "repeat": 1,
                "config": {
                    "factor_1": 4.0,
                    "factor_2": 1.0,
                },
            }
        },
        scheduler=pbt,
        verbose=False)
```

### Describe the problem

tune argument parser expects "trial-resources", but the in variant_generator "trial_resources" is expected.
Thus, resources are not propogated to trials, and the default 1 cpu, 0 gpu is used
### Source code / logs

in tune/variant_generator.py:
 generate_trials(...)
            if "trial_resources" in spec:
                resources = json_to_resources(spec["trial_resources"])
            else:
                resources = None

The resource allocation does not propogate:
> Resources requested: 1/20 CPUs, 0/4 GPUs

After changing the above lines to "trial-resources", I get the correct allocation:
> Resources requested: 4/20 CPUs, 1/4 GPUs

**Comment by ericl** [comment]
Is this 0.4.0 source code? I looked at the pbt_example.py on master, and there is no "trial-resources" field.

**Comment by ericl** [comment]
I think this is working as intended: in the json the syntax is with underscores, and on the command line with dashes (to be consistent with other ray cli params).

**Comment by ericl** [comment]
I guess one weird thing here is that using "trial-resources" silently fails instead of raising an error like other typos, we should detect this case and raise the appropriate error.

**Comment by richardliaw** [comment]
Should be addressed in #2400.
